src/core/lib/gprpp/load_file.h

```diff
@@ -0,0 +1,34 @@
+// Copyright 2022 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GRPC_CORE_LIB_GPRPP_LOAD_FILE_H
+#define GRPC_CORE_LIB_GPRPP_LOAD_FILE_H
+
+#include <grpc/support/port_platform.h>
+
+#include <string>
+
+#include "absl/status/statusor.h"
+
+#include "src/core/lib/slice/slice.h"
+
+namespace grpc_core {
+
+// Loads the content of a file into a slice. add_null_terminator will add a NULL
+// terminator if true.
+absl::StatusOr<Slice> GrpcLoadFile(std::string filename,
```

Comment:
This can just be `LoadFile` - the grpc-ness can be inferred by the grpc_core namespace.

Reply:
Let's actually split this out and land it separately - it should probably have its own test, and we need to update the other 90-odd call sites.

I'd recommend:
1. load_file.{h,cc} PR to land this bit
2. this pr
3. pr to convert grpc_load_file call sites

(1) is likely to stick, (2) has some risk attached, and (3) is work we should do to clean up here

Reply:
I landed a LoadFile PR: https://github.com/grpc/grpc/pull/31176 now.